llvm/include/llvm/Analysis/TargetTransformInfoImpl.h

```diff
@@ -364,7 +364,7 @@ class TargetTransformInfoImplBase {
                               Scale, AddrSpace, /*I=*/nullptr,
                               BaseOffset.getScalable()))
       return 0;
-    return -1;
+    return InstructionCost::getInvalid();
```

Comment:
The fix looks sensible, but the fact all the tests pass means that we were also missing some test coverage. Do you know if it's even possible to expose the invalid cost path in a test?

Reply:
The only caller I've found so far is in `getScalingFactorCost` in LSR. If `isAMCompletelyFolded` returns true for an `LSRUse::Address` kind, `TTI.getScalingFactorCost` is called twice. Right after the calls there is an assert that the cost is valid. That used to be an assert that the cost is positive. You can see this on line 1782 of the diff to LoopStrengthReduce.cpp in https://github.com/llvm/llvm-project/commit/43ace8b5ce07a90b3562bbd0488f88e6290627c4

I think there is an implicit contract that if `isAMCompletelyFolded` returns true, then `getScalingFactorCost` should return a valid cost.

Reply:
Ah I see. That's fine then, and I guess since the LSR code asserts the cost is valid at least we should catch any problems.